### PR TITLE
Autoassign new issues to the Development Github project

### LINKED ISSUE #599: Tracking: Improve Dev Tooling
This is a long-running/permanent tracking issue to link other tooling issues and small tool changes that don't need their own PRs against

### PR DESCRIPTION
**Pull request by Garandor**
As it says on the tin
fix #599 

Signed-off-by: Adam Reif <garandor@manta.network>

## Description

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

**CI workflow changes only, checklist not required**

### CODE REVIEW

File: .github/workflows/assign_new_issue_to_project [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Auto Assign Issues to Projects
+
+on: issues
```
Comment (ghzlatarev) [inline .github/workflows/assign_new_issue_to_project]: We should filter out the PRs, if it doesn't already. I think they are considered issues by github.

File: .github/workflows/assign_new_issue_to_project [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  assign_to_project:
+    runs-on: ubuntu-latest
+    name: Assign topic to Project
+    steps:
+    - name: Assign issues with new-topic label to GitHub Project
+      uses: srggrs/assign-one-project-github-action@1.2.0
+      with:
+        project: 'https://github.com/orgs/Manta-Network/projects/8'
+        column_name: 'No Status'
```
Comment (ghzlatarev) [inline .github/workflows/assign_new_issue_to_project]: Probably better to just rename the `No Status` to `Backlog` instead of having them both.

Comment (Garandor) [reply .github/workflows/assign_new_issue_to_project]: There's a difference.
No status means "needs decision on when"
Backlog means "declared not important for now"

I'd like to keep new issues we haven't talked about yet in a separate category

File: .github/workflows/assign_new_issue_to_project (hunk 1 above)
Comment (Garandor) [reply .github/workflows/assign_new_issue_to_project]: Can't find reference to that in the docs
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues

Let's just try and see if it works, we can always push another fix PR

**Comment by bhgomes** [comment]
@Garandor Does this need to be a `.yml` file?

**Comment by Garandor** [comment]
> @Garandor Does this need to be a `.yml` file?

Thanks, totally didn't notice I forgot the extension. Not sure tbh if it strictly *needs* the extension, but consistency ifself is reason enough to add it.

File: .github/workflows/assign_new_issue_to_project.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Auto Assign Issues to Projects
```
Comment (ghzlatarev) [inline .github/workflows/assign_new_issue_to_project.yml:3]: can you fix the warnings

**Review by ghzlatarev — DISMISSED** [review]

File: .github/workflows/assign_new_issue_to_project.yml (hunk 3 above)
Comment (Garandor) [reply .github/workflows/assign_new_issue_to_project.yml:3]: thanks, done

**Review by ghzlatarev — APPROVED** [review]

File: .github/workflows/assign_new_issue_to_project.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+
+name: Auto Assign Issues to Projects
+
+# yamllint disable-line rule:truthy
+on:
+  issues:
+    types: opened
```
Comment (Dengjianping) [inline .github/workflows/assign_new_issue_to_project.yml:8]: So it doesn't work for PRs, right?

Comment (Garandor) [reply .github/workflows/assign_new_issue_to_project.yml:8]: From what i can read in the docs, it doesn't. See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues
there's a separate `pull_request` event.

If it turns out it does, we'll push a followup PR

**Review by Dengjianping — APPROVED** [review]
LGTM!

**Comment by Garandor** [comment]
There's a difference.
No status means "needs decision on when"
Backlog means "declared not important for now"

**Pull request merged**